include/libstatistics_collector/topic_statistics_collector/received_message_age.hpp

```diff
@@ -88,22 +88,32 @@ struct TimeStamp<M, typename std::enable_if<HasHeader<M>::value>::type>
 
 /**
  * Primary specialization class template until deprecated templated class is phased out
+ * @warning Don't use templated version of the ReceivedMessageAgeCollector, use
+ * libstatistics_collector::ReceivedMessageAgeCollector alias with rmw_message_info_t
+ * parameter in the OnMessageReceived callback
  */
 template<typename T = rmw_message_info_t, typename Enable = void>
 class ReceivedMessageAgeCollector : public TopicStatisticsCollector<T>
 {};
 
 /**
- * Class used to measure the received messsage, tparam T, age from a ROS2 subscriber.
+ * Class used to measure the received message, tparam T, age from a ROS2 subscriber.
  *
  * @tparam T the message type to receive from the subscriber / listener
 */
 template<typename T>
-class ReceivedMessageAgeCollector<
-    T, std::enable_if_t<!std::is_same<T, rmw_message_info_t>::value>>
+class
+  [[deprecated("Don't use templated version of the ReceivedMessageAgeCollector, use"
+  "libstatistics_collector::ReceivedMessageAgeCollector with rmw_message_info_t parameter in the"
+  "OnMessageReceived callback")]]
```

Comment:
could be simplified, and more constructive to introduce the replacement. (the same comment goes to  received_message_period.hpp and topic_statistics_collector.hpp)

```suggestion
  [[deprecated("use libstatistics_collector::ReceivedMessageAgeCollector with rmw_message_info_t instead")]]
```

Reply:
@fujitatomoya I also was considering simplification and reducing the number of words in the comment.  
However, decided to put an explicit warning to not use a templated version of the classes since we can't put the same deprecation warning to the template specializations with the `rmw_message_info_t ` and we would like to get rid of all of them eventually in favor of the non-templated implementation.

Reply:
I would have to have agree with @MichaelOrlov here with keeping 
> "Don't use templated version of the ReceivedMessageAgeCollector, use"
  "libstatistics_collector::ReceivedMessageAgeCollector with rmw_message_info_t"

but I feel it is unnecessary to add information about the callback, since those who previously used the OnMessageReceived will understand that their `rmw_message_info_t` was their custom type `T` and newer users don't need extra context that may confuse them

Reply:
@CursedRock17  I can change it to the 
`"Don't use the templated version of the ReceivedMessageAgeCollector, use libstatistics_collector::ReceivedMessageAgeCollector alias instead"`
Is it Ok?

Reply:
Sounds good, after that am I all good to merge this in, then push to the main branch?

Reply:
@CursedRock17 I've committed https://github.com/CursedRock17/libstatistics_collector/pull/1/commits/47b6b0686d46b0567bddd451d1ea960c8b201df9 changes for simplifying deprecation messages. 
Feel free to merge it on your branch.

As regards to the 
> then push to the main branch?

No, you will not be able to push changes from your branch to the master. 
We will get all approvals and then run CI on the buildfarms to make sure that all dependent packages are fine with our changes on all supported platforms. Only then we can merge it into the master.

Reply:
> We will get all approvals and then run CI on the buildfarms to make sure that all dependent packages are fine with our changes on all supported platforms. Only then we can merge it into the master.

Oh yeah I know that, I just worded it wrong I meant push this to my branch to the one that we are trying to get merged ahead, so that it matches that one. Setting this up right now.